code_generator/GdkCodeGenerator/Templates/UnityGameObjectCommandHandlersGenerator.tt

```diff
@@ -8,33 +8,61 @@
 #>
 <#= generatedHeader #>
 
+using System;
+using System.Collections.Generic;
+using Improbable;
 using Improbable.Gdk.Core;
 using Improbable.Gdk.Core.GameObjectRepresentation;
+using Improbable.Worker;
 using Unity.Entities;
+using UnityEngine;
 
 namespace <#= qualifiedNamespace #>
 {
     public partial class <#= componentDetails.ComponentName #>
     {
         public partial class Requirables
         {
-            [InjectableId(InjectableType.CommandSender, <#= unityComponentDefinition.Id #>)]
-            internal class CommandSenderCreator : IInjectableCreator
+            [InjectableId(InjectableType.CommandRequestSender, <#= unityComponentDefinition.Id #>)]
+            internal class CommandRequestSenderCreator : IInjectableCreator
             {
                 public IInjectable CreateInjectable(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher)
                 {
-                    return new CommandSender(entity, entityManager, logDispatcher);
+                    return new CommandRequestSender(entity, entityManager, logDispatcher);
                 }
             }
 
-            [InjectableId(InjectableType.CommandSender, <#= unityComponentDefinition.Id #>)]
+            [InjectableId(InjectableType.CommandRequestSender, <#= unityComponentDefinition.Id #>)]
             [InjectionCondition(InjectionCondition.RequireNothing)]
-            public class CommandSender : IInjectable
+            public class CommandRequestSender : IInjectable
             {
-                public CommandSender(Entity entity, EntityManager entityManager, ILogDispatcher logger)
+                private Entity entity;
+                private readonly EntityManager entityManager;
+                private readonly ILogDispatcher logger;
+
+                public CommandRequestSender(Entity entity, EntityManager entityManager, ILogDispatcher logger)
                 {
+                    this.entity = entity;
+                    this.entityManager = entityManager;
+                    this.logger = logger;
+                }
 
+<# foreach(var commandDetails in commandDetailsList) { #>
+                public void Send<#= commandDetails.CommandName #>Request(EntityId entityId, <#= commandDetails.RequestType #> request)
+                {
+                    try
+                    {
+                        var ecsCommandRequestSender = entityManager.GetComponentData<CommandSenders.<#= commandDetails.CommandName #>>(entity);
+                        ecsCommandRequestSender.RequestsToSend.Add(new <#= commandDetails.CommandName #>.Request(entityId, request));
+                    }
+                    catch (Exception e)
```

Comment:
What are we expecting this to catch?

Reply:
There is nothing that we are looking for here in particular. Sending commands should always work under normal conditions. Admittedly, I added this in to mirror the approach we have in `ReaderWriterBase`. Happy to remove this if you think it's overkill.